src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -52,37 +52,57 @@ public async Task PrepareJobAsync(IExecutionContext context, List<ContainerInfo>
 
             var response = await ExecuteHookScript(context, input);
             // TODO: Should we throw if response.Context is null or just noop?
-            context.JobContext.Container["id"] = new StringContextData(response.Context.Container.Id);
-            jobContainer.ContainerId = response.Context.Container.Id;
-            context.JobContext.Container["network"] = new StringContextData(response.Context.Container.Network);
-            jobContainer.ContainerNetwork = response.Context.Container.Network;
-            context.JobContext["hook_state"] = new StringContextData(JsonUtility.ToString(response.State));
-
-            // TODO: figure out if we need ContainerRuntimePath for anything
-            // var configEnvFormat = "--format \"{{range .Config.Env}}{{println .}}{{end}}\"";
-            // var containerEnv = await _dockerManager.DockerInspect(executionContext, container.ContainerId, configEnvFormat);
-            // container.ContainerRuntimePath = DockerUtil.ParsePathFromConfigEnv(containerEnv);
-
-            for (var i = 0; i < response.Context.Services.Count; i++)
+            if (response.Context != null)
```

Comment:
Could we use `response?.Context?.[propName]?.` to simplify these `if`s to avoid nesting this deep?